### PR TITLE
OSD-7095 - Adding localprovider tests

### PR DESCRIPTION
**Pull request by ravitri**
### What type of PR is this?
_(test)_

### What this PR does / why we need it?
This PR is to add the tests for `localprovider` package for MUO.

### Which Jira/Github issue(s) this PR fixes?
_Fixes #[OSD-7095](https://issues.redhat.com/browse/OSD-7095)_

### Special notes for your reviewer:
None

### Pre-checks (if applicable):
- [x] Ran `make test` command locally to validate code changes

### CODE REVIEW

File: pkg/localprovider/localprovider_suite_test.go [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package localprovider_test
```
Comment (bmeng) [inline pkg/localprovider/localprovider_suite_test.go:1]: package should be localprovider

File: pkg/localprovider/localprovider_test.go [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+			cfgname: TEST_LOCAL_UPGRADECONFIG_NAME,
+		}
+
+		upgradeConfigList = v1alpha1.UpgradeConfigList{
+			Items: []v1alpha1.UpgradeConfig{
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "uc1",
+					},
+					Spec: v1alpha1.UpgradeConfigSpec{
```
Comment (bmeng) [inline pkg/localprovider/localprovider_test.go:52]: Can we just use the defined spec contains, like:
```
Spec: upgradeConfigSpec,
```
If they have the same values.

File: pkg/localprovider/localprovider_suite_test.go (hunk 1 above)
Comment (ravitri) [reply pkg/localprovider/localprovider_suite_test.go]: Thanks Meng Bo! Done.

File: pkg/localprovider/localprovider_test.go (hunk 2 above)
Comment (ravitri) [reply pkg/localprovider/localprovider_test.go]: Nice suggestion. Would be cleaner. Done.

**Comment by mrbarge** [comment]
One of the behavioural characteristics of the localprovider is that it doesn't return a Upgrade Config spec if the UpgradeConfig's history indicates that it is upgraded. There is the opportunity for having a test around this that is presently lacking.

**Comment by ravitri** [comment]
Thanks a lot Matt! Have added the required test.

**Pull request merged**